src/odemis/gui/model/__init__.py

```diff
@@ -1433,24 +1402,47 @@ def __init__(self, microscope):
+            raise KeyError("Stage has no MD_SAMPLE_CENTERS metadata.")
+        centers = md[model.MD_SAMPLE_CENTERS]
+        layout = sample_positions_to_layout(centers)
+        for row_idx, row in enumerate(layout):
+            for column_idx, name in enumerate(row):
+                scintillator_number = int(name.split()[-1])
+                layout[row_idx][column_idx] = scintillator_number
+        # SAMPLE_SIZES contains the sizes of the scintillators
+        if model.MD_SAMPLE_SIZES not in md:
+            raise KeyError("Stage has no MD_SAMPLE_SIZES metadata.")
+        sample_sizes = md[model.MD_SAMPLE_SIZES]
+        sizes = {}
+        for name, size in sample_sizes.items():
+            scintillator_number = int(name.split()[-1])
+            sizes[scintillator_number] = size
+        # SAMPLE_BACKGROUND contains the minx, miny, maxx, maxy positions of rectangles for
+        # background from bottom-left position with a sample
+        if model.MD_SAMPLE_BACKGROUND not in md:
+            raise KeyError("Stage has no MD_SAMPLE_BACKGROUND metadata.")
+        background = md[model.MD_SAMPLE_BACKGROUND]
+        if set(centers.keys()) != set(sample_sizes.keys()):
+            raise KeyError("MD_SAMPLE_CENTERS and MD_SAMPLE_SIZES metadata should have "
+                "the same keys.")
 
         # Initialize attributes related to the sample carrier
-        #  * .scintillator_size (float, float): size of one scintillator in m
-        #  * .scintillator_positions (dict: 1 <= int <= 9 --> (float, float)): positions in stage coordinates
+        #  * .scintillator_sizes (dict: int --> (float, float)): size of scintillators in m
+        #  * .scintillator_positions (dict: int --> (float, float)): positions in stage coordinates
         #  * .scintillator_layout (list of list of int): 2D layout of scintillator grid
         #  * .background (list of ltrb tuples): coordinates for background overlay,
         #    rectangles can be displayed in world overlay as grey bars, e.g. for simulating a sample carrier
-        self.scintillator_size = carrier_params["scintillator_size"]
-        self.scintillator_positions = {}  # dict: 1 <= int <= 9 --> (float, float)
-        for num, offset in carrier_params["scintillator_offsets"].items():
-            self.scintillator_positions[num] = (minx + offset[0], miny + offset[1])
-        self.scintillator_layout = carrier_params["layout"]
+        self.scintillator_sizes = sizes
+        self.scintillator_positions = {}
+        for name, center in centers.items():
+            scintillator_number = int(name.split()[-1])
+            self.scintillator_positions[scintillator_number] = (minx + center[0], miny + center[1])
+        self.scintillator_layout = layout
         self.background = []
-        for rect in carrier_params["background"]:
+        for rect in background:
             self.background.append((minx + rect[0], miny + rect[1], minx + rect[2], miny + rect[3]))
```

Comment:
Please catch any error here and report as an error in  SAMPLE_BACKGROUND. One of the typical error that could happen is that a list is too short , or that a number is entered like `1e-6` and interpreted as a string. Such error coming from the GUI will be very hard to understand by the installation engineer if there is not a nice explanation!